src/util/STYCRVHelper.sol

```diff
@@ -0,0 +1,161 @@
+   
+    uint256 public constant scale = 1e18;
+    ISTYCRV public constant vault =
+        ISTYCRV(0x27B5739e22ad9033bcBf192059122d163b60349D); // st-yCRV
+    uint256 public maxLoss = 1; // 0.01% [BPS]
+    IERC20 public constant underlying =
+        IERC20(0xFCc5c47bE19d06BF83eB04298b026F81069ff65b); // yCRV
+
+    IMarket public constant market =
+        IMarket(0x27b6c301Fd441f3345d61B7a4245E1F823c3F9c4); // DOLA market
+
+    struct Permit {
+        uint256 deadline;
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+    }
+
+    constructor() Ownable(msg.sender) {
+        underlying.approve(address(vault), type(uint256).max);
+        IERC20(address(vault)).approve(address(market), type(uint256).max);
+    }
+
+    function setMaxLoss(uint256 _maxLoss) external onlyOwner {
+        maxLoss = _maxLoss;
+    }
+
+    /// @notice Transforms underlying to collateral
+    /// @param _value Amount of underlying to transform
+    /// @param _helperData Optional helper data in case the collateral needs to be transformed
+    /// @return collateralAmount Amount of collateral received
+    function transformToCollateral(
+        uint256 _value,
+        bytes calldata _helperData
+    ) external nonReentrant returns (uint256 collateralAmount) {
+        if(_value > vault.availableDepositLimit()) revert DepositLimitExceeded();
+
+        underlying.transferFrom(msg.sender, address(this), _value);
+
+        uint256 estimateAmount = assetToCollateral(_value);
+
+        // Transform underlying to collateral
+        collateralAmount = vault.deposit(_value, msg.sender);
+        
+        if (collateralAmount < estimateAmount)
+            revert DepositFailed(estimateAmount, collateralAmount);
+    }
+
+    /// @notice Transforms collateral to underlying
+    /// @param _value Amount of collateral to transform
+    /// @param _helperData Optional helper data in case the collateral needs to be transformed
+    /// @return underlyingAmount Amount of underlying received
+    function transformFromCollateral(
+        uint256 _value,
+        bytes calldata _helperData
+    ) external nonReentrant returns (uint256 underlyingAmount) {
+        if (_value > vault.maxAvailableShares()) revert NotEnoughShares();
+
+        IERC20(address(vault)).transferFrom(msg.sender, address(this), _value);
+
+        uint256 estimateAmount = collateralToAsset(_value);
+
+        underlyingAmount = vault.withdraw(_value, msg.sender, maxLoss);
+
+        if (underlyingAmount < estimateAmount) 
+            revert WithdrawFailed(estimateAmount, underlyingAmount);
+    }
+
+    /// @notice View function to calculate collateral amount from asset amount
+    /// @param assetAmount Amount of asset to transform
+    /// @return collateralAmount Amount of collateral received
+    function assetToCollateral(
+        uint assetAmount
+    ) public view returns (uint collateralAmount) {
+        return assetAmount * vault.totalSupply() / getFreeFunds();
```

Comment:
This needs an extra check against total supply to make sure its not 0. like this...
```
uint totalSupply = IVault(vault).totalSupply();
if (totalSupply > 0) {
     return amount * totalSupply / calculateFreeFunds(vault);
}
return amount;
```